src/js/heuristicblocking.js

```diff
@@ -125,12 +127,115 @@ HeuristicBlocker.prototype = {
       return {};
     }
 
-    // ignore if there are no tracking cookies
-    if (!hasCookieTracking(details, origin)) {
+    // check if there are tracking cookies
+    if (hasCookieTracking(details, origin)) {
+      this._recordPrevalence(fqdn, origin, tabOrigin);
       return {};
     }
 
-    this._recordPrevalence(fqdn, origin, tabOrigin);
+    let self = this;
+
+    // check if this request is a cookie-syncing pixel
+    chrome.cookies.getAll({}, function(cookies) {
+      self.pixelCookieSyncAccounting(details, cookies);
+    });
+  },
+
+  /**
+   * Checks for cookie syncing: requests to third-party domains that include
+   * high entropy data from first-party cookies.
+   *
+   * TODO: is it OK to call this from heuristicBlockingAccounting?
+   *
+   * @param details are those from onBeforeSendHeaders
+   * @param cookies are the result of chrome.cookies.getAll()
+   * @returns {*}
+   */
+  pixelCookieSyncAccounting: function (details, cookies) {
+    // only interested in images
+    if (details.type != 'image') {
+      return false;
+    }
+
+    let fqdn = (new URI(details.url)).host,
```

Comment:
We already computed this in the parent function ... Should probably pass all this existing stuff as parameters.